src/ott/problems/linear/barycenter_problem.py

```diff
@@ -176,6 +176,55 @@ def tree_flatten(self) -> Tuple[Sequence[Any], Dict[str, Any]]:  # noqa: D102
   @classmethod
   def tree_unflatten(  # noqa: D102
       cls, aux_data: Dict[str, Any], children: Sequence[Any]
-  ) -> "BarycenterProblem":
+  ) -> "FreeBarycenterProblem":
     y, b, weights = children
     return cls(y=y, b=b, weights=weights, **aux_data)
+
+
+@jax.tree_util.register_pytree_node_class
+class FixedBarycenterProblem:
+  """Fixed-support Wasserstein barycenter :cite:`janati:20a`.
+
+  Args:
+    geom: geometry object.
+    a:  batch of histograms of shape ``[batch, num_a]``.
+    weights: positive weights in the probability simplex.
+  """
+
+  def __init__(
+      self,
+      geom: jnp.ndarray,
+      a: Optional[jnp.ndarray] = None,
+      weights: Optional[jnp.ndarray] = None,
+      **kwargs: Any,
+  ):
+    self.geom = geom
+    self.a = a
+    self._weights = weights
+
+  @property
+  def num_measures(self) -> int:
+    """Number of measures."""
+    return self.a.shape[0]
```

Comment:
This will cause issues when `self.a = None`.

Reply:
thanks! in fact `a` cannot be `None`, this wouldn't make sense, so removed the `optional`